### DISCUSSION TITLE
https://angular.love/angular-router-everything-you-need-to-know-about

### DISCUSSION THREAD

**Discussion by giscus** [HouseOfAngular/angular-love#368] [General]
# https://angular.love/angular-router-everything-you-need-to-know-about

Angular.love - a place for all Angular enthusiasts created to inspire and educate.

https://angular.love/angular-router-everything-you-need-to-know-about

**Comment by raven1152** [comment]
This article has really expanded my angular knowledge, of the router, and other related topics, plus a few gold nuggets, like the use of the inject function.  I would love if you were to write a similar article covering everything you need to know about reactive forms, or even just the form builder service.  i have been working with several ideas for how to implement forms in a large application, including using injectable services that provide the form and all of it's validation, for a given data model, including custom patch functions that deal with key parts of a larger model.

**Comment by Und3rMySk1n** [comment]
Hi! Thank you for useful and easy to understand article. I guess, you have a misprint in canDeactivate example:

```
canActivate: [canDeactivate] 
```

Not critical, but easy to fix. Thanks for sharing your experience)
